test/unit/model/define.test.js

```diff
@@ -35,15 +35,60 @@ describe(Support.getTestDialectTeaser('Model'), () => {
         current.define('foo', {
           id: DataTypes.INTEGER,
         });
-      }).to.throw('A column called \'id\' was added to the attributes of \'foos\' but not marked with \'primaryKey: true\'');
+      }).to.throw('An attribute called \'id\' was defined in model \'foos\' but not marked as a primaryKey. This is likely to be an error, which can be fixed by setting its \'primaryKey\' option to true. If this is intended, explicitly set its \'primaryKey\' option to false');
 
       expect(() => {
         current.define('bar', {
           id: {
             type: DataTypes.INTEGER,
           },
         });
-      }).to.throw('A column called \'id\' was added to the attributes of \'bars\' but not marked with \'primaryKey: true\'');
+      }).to.throw('An attribute called \'id\' was defined in model \'bars\' but not marked as a primaryKey. This is likely to be an error, which can be fixed by setting its \'primaryKey\' option to true. If this is intended, explicitly set its \'primaryKey\' option to false');
+    });
+
+    it('should allow model definition without PK', () => {
+      const Model = current.define('User', {}, {
+        noPrimaryKey: true,
+      });
+
+      expect(Model.rawAttributes).not.to.have.property('id');
+    });
+
+    it('should allow model definition without PK with noPrimaryKey is set to true', () => {
+      const Model = current.define('User', {}, {
+        noPrimaryKey: true,
+      });
+
+      expect(Model.rawAttributes).not.to.have.property('id');
+    });
+
+    it('should add the default `id` field PK if noPrimary is set', () => {
+      const Model = current.define('User', {});
+
+      expect(Model.rawAttributes).to.have.property('id');
+    });
+
+    it('should not add the default `id` field PK if PK has been defined manually', () => {
+      const Model = current.define('User', {
+        id: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+        },
+      });
+
+      expect(Model.rawAttributes).to.have.property('id');
```

Comment:
for this test to work, the PK should be named something different than 'id', otherwise we can't test that 'id' hasn't been added

```suggestion
      const Model = current.define('User', {
        customId: {
          type: DataTypes.INTEGER,
          primaryKey: true,
        },
      });

      expect(Model.rawAttributes).not.to.have.property('id');
```